wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -106,45 +106,47 @@ private[wallet] trait UtxoHandling extends WalletLogger {
           case None =>
             logger.warn(
               s"Given txos exist in block (${blockHash.hex}) that we do not have! $txos")
-            txos
+            Vector.empty
           case Some(confs) =>
-            txos.map { txo =>
+            txos.flatMap { txo =>
               txo.state match {
                 case TxoState.ImmatureCoinbase =>
                   if (confs > Consensus.coinbaseMaturity) {
                     if (confs >= walletConfig.requiredConfirmations) {
```

Comment:
So with coinbase txs we need 100 confirmations + the require confirmations? I guess htis does seem right otherwise we can have a reorg happen that causes the coinbase utxo be pending again. I hadn't though of this corner case before :/

Reply:
This is just incase someone sets `walletConfig.requiredConfirmations` greater than 100 `Consensus.coinbaseMaturity`